### ISSUE TITLE
support for string interpolations in quasiquotes

### ISSUE THREAD

**Issue by xeno-by** [scalameta/scalameta#251]
The quasiquote spec includes some wishful thinking like `q""" $name"$${..$exprs}" """` to works with string interpolations and insert/extract their parts. Since it doesn't capture static parts and has never been implemented, I'm now gonna remove it and honestly say that string interpolation isn't implemented.

**Comment by xeno-by** [comment]
In the meanwhile, use `scala.meta.internal.ast.Term.Interpolate` and `scala.meta.internal.ast.Pat.Interpolate` until we don't fix this bug.

**Comment by ysusuk** [comment]
Hi @xeno-by If you point me to the right direction i'll be glad to try to fix it!

**Comment by xeno-by** [comment]
Hi @ysusuk! Thanks for your offer.

This is a tricky one, because at the moment it's unclear for me not just how to implement this, but in general what syntax we should use for string interpolations in quasiquotes. Do you maybe have ideas here?

If you'd like to start with something more approachable, I can offer you a couple other issues. Please let me know.

**Comment by ysusuk** [comment]
Hi @xeno-by!
I'll be happy to start with something more approachable=) thank you!

**Comment by olafurpg** [comment]
@ysusuk You might be able to find something interesting labeled as "Hackathon" https://github.com/scalameta/scalameta/issues?q=is%3Aopen+is%3Aissue+label%3AHackathon

Those issues are mostly low hanging fruit and we've tried to include useful links in the descriptions.

**Comment by xeno-by** [comment]
@olafurpg and I had a discussion, and we decided to pronounce quasiquotes feature-complete.

There are still some open issues (#46, #223, #251, #280 and others), but they all have two things in common: 1) they require non-trivial design work to be fixed, 2) they aren't popular at all. Given our recent decision to start avoiding unwarranted breaking changes in 2.0, we'll consider quasiquotes good enough at this point and will close all outstanding feature requests.

We will still be fixing bugs (in fact, I've fixed about a dozen of such bugs for 2.0), but without strong evidence our design efforts will be directed elsewhere.
